torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -2904,8 +2907,26 @@ def _persistent_reduction_configs(
     if "y" in size_hints:
         pass
     # TODO(jansel): we should be able to improve these heuristics
-    elif reduction_hint == ReductionHint.INNER and rnumel >= 256:
-        configs = configs[:1]
+    elif reduction_hint == ReductionHint.INNER:
+        if rnumel > 1024:
+            configs = configs[:1]
+        else:
+            x_block = 8 if rnumel < 1024 else 4
+            if xnumel // x_block < 128 or (loads_and_stores >= 5 and rnumel >= 256):
+                # If loads/stores greater than 5, a lot of register pressure
+                # rnumel < 256 means no vectorized loads if we split up r dim
```

Comment:
How do you come up with the threshold 256 here regarding vectorized load/store? Does it assume specific dtype and num_warps?